src/qt/sendcoinsdialog.cpp

```diff
@@ -889,17 +885,29 @@ void SendCoinsDialog::coinControlUpdateLabels()
     }
 }
 
-SendConfirmationDialog::SendConfirmationDialog(const QString& title, const QString& text, const QString& informative_text, const QString& detailed_text, int _secDelay, QWidget* parent)
-    : QMessageBox(parent), secDelay(_secDelay)
+SendConfirmationDialog::SendConfirmationDialog(const QMessageBox::Icon icon, const QString& title, const QString& text, const QMessageBox::StandardButton yes_button, const QString &yes_button_text, const QMessageBox::StandardButton cancel_button, const QString& informative_text, const QString& detailed_text, int _secDelay, QWidget *parent)
+    : QMessageBox(parent),
+    m_yes_button_text(yes_button_text),
+    secDelay(_secDelay)
 {
-    setIcon(QMessageBox::Question);
+    setIcon(icon);
     setWindowTitle(title); // On macOS, the window title is ignored (as required by the macOS Guidelines).
     setText(text);
     setInformativeText(informative_text);
     setDetailedText(detailed_text);
-    setStandardButtons(QMessageBox::Yes | QMessageBox::Cancel);
-    setDefaultButton(QMessageBox::Cancel);
-    yesButton = button(QMessageBox::Yes);
+    setStandardButtons(yes_button | cancel_button);
```

Comment:
Does this line actually do anything, if we subsequently remove both buttons and re-add them? Can we just call addButton twice to the same effect, and not have to remove them first?

Reply:
@gwillen I think that has something to do with ordering: https://github.com/bitcoin/bitcoin/pull/17463#discussion_r346892722